src/Database/Database.php

```diff
@@ -4335,8 +4377,12 @@ public function decode(Document $collection, Document $document, array $selectio
                 }
             }
 
-            if (empty($selections) || \in_array($key, $selections) || \in_array('*', $selections)) {
-                $document->setAttribute($key, ($array) ? $value : $value[0]);
+            if (empty($selections) || \in_array($key, $selections) || \in_array('*', $selections) || \in_array($key, ['$createdAt', '$updatedAt'])) {
```

Comment:
Let's break this into new lines to make it easier to read:

```suggestion
            if (
                empty($selections) 
                || \in_array($key, $selections) 
                || \in_array('*', $selections) 
                || \in_array($key, ['$createdAt', '$updatedAt'])
            ) {
```